test/SignatureSpec.hs

```diff
@@ -0,0 +1,42 @@
+{-# OPTIONS_GHC -Wno-unused-record-wildcards #-}
+
+module SignatureSpec (spec) where
+import Test.Hspec
+import IOHK.Certification.SignatureVerification
+import Data.Aeson
+import Plutus.Certification.API
+import Text.RawString.QQ(r)
+
+(Just laceObject) = decode @LoginBody [r|{
+  "address": "addr_test1qr9g2tnvpgc264tvnjkcpqfeulvmuu2k26enq25yszwl5yw6gvppa4mz9sayrvfklds5dqq44nl36xkng33kz738g0hqj0adak",
+  "key": "a5010102583900ca852e6c0a30ad556c9cad808139e7d9be715656b3302a84809dfa11da43021ed7622c3a41b136fb61468015acff1d1ad34463617a2743ee03272006215820d3be7240cd3f131316a2489c609ff4d7a75732834bd120f34618afc66a236f16",
+  "signature": "845882a3012704583900ca852e6c0a30ad556c9cad808139e7d9be715656b3302a84809dfa11da43021ed7622c3a41b136fb61468015acff1d1ad34463617a2743ee6761646472657373583900ca852e6c0a30ad556c9cad808139e7d9be715656b3302a84809dfa11da43021ed7622c3a41b136fb61468015acff1d1ad34463617a2743eea166686173686564f458d45369676e2074686973206d65737361676520696620796f752061726520746865206f776e6572206f662074686520616464725f74657374317172396732746e7670676332363474766e6a6b6370716665756c766d7575326b3236656e71323579737a776c357977366776707061346d7a397361797276666b6c64733564717134346e6c3336786b6e6733336b7a373338673068716a306164616b20616464726573732e200a2054696d657374616d703a203c3c313639333935383634313e3e200a204578706972793a203630207365636f6e64735840511b7f3b847ad1e5b8ffac44ff4af5f5cbf1f4dda3d4d179dbc5c9ce1626aaca0118795fa41bdd0a4440473a609bfe47f6b4c23bd34b0a9bf8b7a56c6e120009"
+}|]
+
+(Just numyObject) = decode @LoginBody [r|{
+  "address": "addr_test1qr9g2tnvpgc264tvnjkcpqfeulvmuu2k26enq25yszwl5yw6gvppa4mz9sayrvfklds5dqq44nl36xkng33kz738g0hqj0adak",
+  "key": "a4010103272006215820d3be7240cd3f131316a2489c609ff4d7a75732834bd120f34618afc66a236f16",
+  "signature": "845846a201276761646472657373583900ca852e6c0a30ad556c9cad808139e7d9be715656b3302a84809dfa11da43021ed7622c3a41b136fb61468015acff1d1ad34463617a2743eea166686173686564f458d45369676e2074686973206d65737361676520696620796f752061726520746865206f776e6572206f662074686520616464725f74657374317172396732746e7670676332363474766e6a6b6370716665756c766d7575326b3236656e71323579737a776c357977366776707061346d7a397361797276666b6c64733564717134346e6c3336786b6e6733336b7a373338673068716a306164616b20616464726573732e200a2054696d657374616d703a203c3c313639333935373436313e3e200a204578706972793a203630207365636f6e64735840a2559f6b3aaed7dd920c8d647c118758fcf7d958279d0291271bfe77150c8cc3faac5367879236abbb325bd6361f06f5d59e8178feed8406ba3f67140abae504"
+}|]
+
+spec :: SpecWith ()
+spec =
+  describe "Signature verification" $ do
+
+    it "successfully validates Lace signature" $ do
+      let LoginBody{..} = laceObject
+      verifyCIP30Signature key signature Nothing (Just $ Bech32Address address) `shouldBe` Right True
+
+    it "successfully validates Numy signature" $ do
+      let LoginBody{..} = numyObject
+      verifyCIP30Signature key signature Nothing (Just $ Bech32Address address) `shouldBe` Right True
+
+
```

Comment:
In the future, also try some test where we expect to fail

Reply:
You are right. In the other PR's you'll see that too. But in this case, I was rushing it also because normally anything you would alter by hand will fail

Reply:
I added some extra tests